bliss/surveys/dc2.py

```diff
@@ -30,94 +29,55 @@ def map_nested_dicts(cur_dict, func):
     return func(cur_dict)
 
 
-class DC2Dataset(Dataset):
-    def __init__(self, split_files_list, in_memory: bool) -> None:
-        super().__init__()
-        self.split_files_list = split_files_list
-        self.in_memory = in_memory
-        self.in_memory_data = []
-        if self.in_memory:
-            logger = logging.getLogger("DC2")
-            logger.warning("WARNING: you are using in-memory DC2; it takes time to load")
-            for split_file in self.split_files_list:
-                with open(split_file, "rb") as sf:
-                    split = torch.load(sf)
-                    self.in_memory_data.append(split)
-
-    def __len__(self):
-        return len(self.split_files_list)
-
-    def __getitem__(self, idx):
-        if not self.in_memory:
-            with open(self.split_files_list[idx], "rb") as split_file:
-                split = torch.load(split_file)
-        else:
-            split = self.in_memory_data[idx]
-        return split
+def split_list(ori_list, sub_list_len):
+    return [ori_list[i : (i + sub_list_len)] for i in range(0, len(ori_list), sub_list_len)]
 
 
-class DC2(Survey):
+class DC2Dataset(CachedSimulatedDataset):
```

Comment:
Would you rename `CachedSimulatedDataset` and `DC2Dataset` to `CachedSimulatedDataModule` and `DC2DataModule`, respectively? I realize calling them datasets is sort of confusing now that they inherit from `DataModule` and we have other classes that inherit from `Dataset`.

Reply:
yeah, i can handle it